### ISSUE TITLE
License Agreement.

### ISSUE THREAD

**Issue by Mempler** [GrowtopiaNoobs/GrowtopiaServer#1066]
Please guys, follow the License Agreement of this Repository.

stop closing your source as long as @GrowtopiaNoobs doesn't say anything else.
so please, everybody. if you Develope your server make sure to make EVERYTHING open source, otherwise there can be a DMCA request on your server by me!

I'm getting really mad atm that people are screaming "THIS CODE WAS LEAKED OF A SERVER" which is actually supposed to be open source anyway as it's License Agreement!

https://tldrlegal.com/license/gnu-affero-general-public-license-v3-(agpl-3.0)

**Comment by cmdocmd** [comment]
So basically u saying we code hard while u are sleeping or having fun then u tell us to give u codes HAHA in ur dreams

**Comment by Mempler** [comment]
No, what i am saying is that you agreed the license and if you don't follow it, i'll dmca your server. idc about the code as i never made a server myself. *why should i?* i just wanted to enforce the license agreement. this is my final warning to EVERYONE who has setup a server.

**Comment by WoofHax** [comment]
I hope this Project is closed

**Comment by cmdocmd** [comment]
@Mempler Okay u said "i never made a server myself" then why did u even open this issue just close it and who care about private server can open an issue LOL

**Comment by iProgramMC** [comment]
We will never give away our hard earned findings to indonesian noobies who can't code properly!

**Comment by iProgramMC** [comment]
The license doesn't say anything about modifications to the original source. Also, what if we made our OWN server? What would you do?!

**Comment by playingo** [comment]
@iProgramMC we technically did already

**Comment by Mempler** [comment]
@iProgramMC 
https://github.com/GrowtopiaNoobs/GrowtopiaServer/blob/master/LICENSE#L42-L48
do i have to say more ?

**Comment by ardakurucim** [comment]
@Mempler nobody will share it for sure

**Comment by Mempler** [comment]
@ardakurucim that's why i started to DMCA servers.

**Comment by ardakurucim** [comment]
@Mempler i dont think dmca will care of it

**Comment by Mempler** [comment]
Yes, they will. it's US law && Policy of most Servers. ever read those ?

**Comment by Mempler** [comment]
and @iProgramMC if you made your own server from scratch and you have proof, i would do nothing.

**Comment by iProgramMC** [comment]
Where are you going to get the money to DMCA claim servers anyway?

**Comment by GrowtopiaNoobs** [comment]
I was thinking alot about license choice for this server. And I choose the most opensource one - AGPL, so everyone should release their code if they share the server as exe file or use it for running their own server for public.

Well you might ask why I haven't choosen another one? I could just choose GPL, so that people are forced to only share source, when they share the binary file - this could be nice to prevent uploading files which contain viruses, because people could compile themself from source. Or just release it under MIT or something and don't care.

I have released it under AGPL in hope, that community will share their modifications and maintain the proejct. I have quited managing server, because I just came from school and was working all my time to make it more stable, figure more about gt protocol and add more features. Well, it might be hard to imagine, but if you go for month like this, you are in constant stress, you are even in school checking if server is running, trying to look that community around server is fine and happy. I was running it completly for free - not paid items or stuff like current GTPS, so I was trying real hard to get VPS from someone and make sure that he won't steal all players passwords or IPs.

I was really stressed, VPS expired. I got offer to sell source code for not bad money, but I haven't accepted it, beucase I knew community would want to run their own servers alot. So I have released my code, stopped working on it to get free time and left all work to community - all people here on github.

You might decide to not release your modifications and not to help others. You probably won't get DMCA claim, but I can't promise to you that you won't. Decide yourself what you will do.

**Comment by flossboss** [comment]
@GrowtopiaNoobs I agree 👍

**Comment by Mempler** [comment]
That's really well written.

**Comment by cucksucks** [comment]
Does this mean our source code has to be public, like on a file upload? Or
like a pastebin?

**Comment by Mempler** [comment]
No, as a Github Repository with the same license. just fork and update your source via `git push`
https://git-scm.com/docs/git-push

**Comment by IndogLelele** [comment]
lolololol

**Comment by ama6nen** [comment]
dmca something breaking like 5 parts of ubisoft tos & with no verifiable proof that something is violating copyright lol yeah ok

**Comment by iProgramMC** [comment]
If you want to be at least a little bit protected release your server code under the same license so you are under the GTS umbrella. You can DMCA strike anyone who uses your server code in theirs without your permission if you think it's wrong.

**Comment by Mempler** [comment]
wrong, you can't DMCA without a reason. you can publish your code. it's getting licensed under AGPL 3 anyway as GTS license forces that. you CAN'T DMCA a server if the code is public! unless they have changes that they haven't published yet. i would Recommend to you all to read the AGPL 3 License.
